### PR TITLE
Ignore null points for line charts

### LINKED ISSUE #418: [places] skip null points
E.g. health outcomes and behaviors chart in MTV (when shown as a timeline)

Don't treat null as 0

image

### PR DESCRIPTION
**Pull request by beets**
Also fix dots only drawn for one line chart

image

### CODE REVIEW

**Comment by beets** [comment]
Here's another possible treatment

image

image

File: static/js/chart/draw.ts [hunk 1]
```diff
@@ -485,22 +485,32 @@ function drawLineChart(
...
+
+    const line = d3
       .line()
+      .defined((d) => d[1] !== null)
       .x((d) => xScale(d[0]))
       .y((d) => yScale(d[1]));
 
-    if (shouldAddDots) {
-      line = line.curve(d3.curveMonotoneX);
+    if (hasNullValues) {
```
Comment (shifucun) [inline static/js/chart/draw.ts:507]: This looks nice! Can you explain a bit more how this does the magic to use solid lines and dotted lines together? This is fancy d3 :)

**Comment by beets** [comment]
Added warning:
image

Though I don't have a good way to know if there wasn't a fill line if the missing data is at the ends:
image

**Comment by beets** [comment]
Fixed the second chart (will not show warning if there are only missing data on either end):

image

**Review by shifucun — APPROVED** [review]
very nice change!

**Pull request merged**